include/ginkgo/c_api.h

```diff
@@ -0,0 +1,851 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Copyright (c) 2017-2023, the Ginkgo authors
+#ifndef C_API_H
+#define C_API_H
```

Comment:
```suggestion
#ifndef GKO_C_API_H
#define GKO_C_API_H
```
Maybe better to prefix with GKO to avoid potential collisions.